src/content.js

```diff
@@ -174,11 +188,78 @@ function apply(options, insertionPoint) {
 	}
 }
 
+function readAllChunks(readableStream) {
+	const reader = readableStream.getReader();
+	const chunks = [];
+
+	return pump();
+
+	function pump() {
+		return reader.read().then(({value, done}) => {
+			if (done) {
+				return chunks;
+			}
+
+			chunks.push(value);
+			return pump();
+		});
+	}
+}
+
+function requestPage(cookies, number) {
+	return fetch(`https://github.com/dashboard/index/${number}?utf8=%E2%9C%93`, {
+		credentials: 'include',
+		Cookie: cookies,
+		headers: {
+			'X-Requested-With': 'XMLHttpRequest'
+		}
+	})
+		.then(data => {
+			const streem = data.body;
+
+			return readAllChunks(streem)
+				.then(data => data
+					.map(chunk => new TextDecoder('utf-8').decode(chunk))
+					.join('')
+				);
+		});
+}
+
+function preloadPages(options) {
+	const parsedCookies = JSON.parse(options.cookies);
+
+	if (parsedCookies) {
+		const serializedCookies = Object.keys(parsedCookies)
+			.reduce((res, key) => `${res}${key}=${parsedCookies[key]}; `, '');
+
+		Promise.all(
+			Array(options.preloadPagesCount - 1)
+				.fill(null)
+				.map((t, n) => requestPage(serializedCookies, n + 1))
+		)
+			.then(data => {
+				const form = document.querySelector('.ajax-pagination-form');
+				const preloadedNews = document.querySelector('.ghcf-holder');
+
+				const fragment = document.createRange().createContextualFragment(
+					data.join('').replace(/<form[\s\S]*?<\/form>/g, '')
+				);
+
+				preloadedNews.appendChild(fragment);
+				preloadedNews.appendChild(apply(options));
+
+				form.action = form.action.replace(/\/\d$/, '/' + (options.preloadPagesCount));
+			});
+	}
+}
+
```

Comment:
The empty feed in the screenshot is a separate issue, I create it here so maybe I'll fix it separately: https://github.com/bfred-it/github-clean-feed/issues/7

> generally abandon the standard GitHub's render.

That's a good point, worth discussing here: https://github.com/bfred-it/github-clean-feed/issues/8